src/throttler.guard.ts

```diff
@@ -0,0 +1,169 @@
+import { CanActivate, ExecutionContext, Inject, Injectable, ContextType } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import * as md5 from 'md5';
+import { ThrottlerStorage } from './throttler-storage.interface';
+import {
+  THROTTLER_LIMIT,
+  THROTTLER_OPTIONS,
+  THROTTLER_SKIP,
+  THROTTLER_TTL,
+} from './throttler.constants';
+import { ThrottlerException, ThrottlerWsException } from './throttler.exception';
+import { ThrottlerOptions } from './throttler.interface';
+
+@Injectable()
+export class ThrottlerGuard implements CanActivate {
+  storageService: ThrottlerStorage;
+
+  constructor(
+    @Inject(THROTTLER_OPTIONS) private readonly options: ThrottlerOptions,
+    @Inject(ThrottlerStorage) storageService: ThrottlerStorage,
+    private readonly reflector: Reflector,
+  ) {
+    this.storageService = options.storage || storageService;
```

Comment:
@kkoomen what was the reason for this line again? As we set `ThrottlerStorage` in the `ThrottlerCoreModule` according to if an option is passed or not, so this seems like overkill to me.

Reply:
Yeah, you pointed this out in a previous PR, but I haven't had time to fix it and forgot to discuss this with you. If you can confirm locally that removing that line still works then you can remove it 👌🏻